jetty-deploy/src/main/java/org/eclipse/jetty/deploy/DeploymentManager.java

```diff
@@ -237,6 +238,14 @@ protected void doStart() throws Exception
         {
             startAppProvider(provider);
         }
+
+        if (onStartupErrors != null)
+        {
+            RuntimeException mex = new RuntimeException("Failed to successfully start App Providers (See log for details)");
+            onStartupErrors.forEach((cause) -> mex.addSuppressed(cause));
+            throw mex;
+        }
+
```

Comment:
This change will always fail the DeploymentManager startup, and thus the server startup if one of the webapps fail to deploy. Do we want to make it configurable so that the DeploymentManager and thus Server will start, even if a webapp fails to deploy? Or is this what the WebAppContext.setThrowUnavailableOnStartupException is doing?

Reply:
The exception will not flow out of the WebAppContext unless it has `.setThrowUnavailableOnStartupException(true)`
See `WebAppContext.doStart()` ...

https://github.com/eclipse/jetty.project/blob/205c4dc498d16431a3bd81ddcaf52cd45f72d4d0/jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebAppContext.java#L551-L559

Reply:
I wanted this to just flow out of the `WebAppContext.doStart()` all the way up to the `Server.doStart()`, but that means we would have to make `Scanner` support this too, which would have an undesirable impact on other components that rely on `Scanner`.

I'll go ahead and add two more testcases where we have a `badapp.war`.
1. One that has a context xml deployable specifically with `<Set name="throwUnavailableOnStartupException">false</Set>`.
2. One that has no context xml deployable and is using default behavior.

Reply:
I've added the two new testcases and also am using MultiException now.
@janbartel can I get a review of this PR from you?